### PR TITLE
CAD-3257 generator service:  absorb defaultGeneratorScriptFn

### PR DESCRIPTION
**Pull request by deepfire**
This moves the (configuration -> generator run script) translation function into `tx-generator-service.nix` from the workbench into the NixOS service definition.

### CODE REVIEW

**Review by MarcFontaine — APPROVED** [review]
LGTM

**Review by MarcFontaine — APPROVED** [review]
LGTM

File: nix/svclib.nix [hunk 1]
```diff
@@ -171,19 +171,13 @@ let
...
-    listOpt = opt (types.listOf types.attrs);
-    enumOpt = set: default: description:
-      mkOption { inherit default description;
-                 type = types.enum set; };
-  };
+  serviceDeclarationLib = with pkgs.lib; with types;
+    types // rec {
+      opt    = type: default: description:
+               mkOption { inherit type default description; };
+      mayOpt = type: description:
```
[the remark this answers is not in the document]
Comment (deepfire) [reply nix/svclib.nix:180]: `mayOpt` denotes `null`-ability.

Pushed doc comments : -)

**Pull request merged**